packages/block-library/src/embed/util.js

```diff
@@ -198,6 +198,14 @@ export function getClassNames(
 		) {
 			const potentialRatio = ASPECT_RATIOS[ ratioIndex ];
 			if ( aspectRatio >= potentialRatio.ratio ) {
+				// Evaluate the difference between actual aspect ratio and closest match.
+				// If the difference is too big, do not scale the embed according to aspect ratio.
+				const ratioDiff = aspectRatio - potentialRatio.ratio;
+				if ( ratioDiff > 0.1 ) {
```

Comment:
We want to find a balance with this differential where actual matches are handled with nice aspect ratio scaling, but incorrect ones are not. `0.05` was also fine for identifying a 4:3 YouTube video correctly, but bailing on the Spotify one.